packages/react-components/src/components/chart/hooks/useChartsLegend.tsx

```diff
@@ -45,11 +45,15 @@ const useChartsLegend = ({
   ]);
   const [items, setItems] = useState<Array<object>>([]);
   useEffect(() => {
-    const tcColumnDefinitions = graphic.map((g) => ({
-      id: g.id as string,
-      header: getHeaderNode(g),
-      cell: (e: { [x: string]: string }) => e[g.id as string],
-    }));
+    const tcColumnDefinitions = graphic.map((g) => {
+      const id = g.id as string;
```

Comment:
Looks like this comes from `OptionId` in e-charts - https://github.com/apache/echarts/blob/master/src/util/types.ts#L626. Better to check/validate or cast this to a string rather than assume. I know this isn't directly your changes but just something I noticed. Accidentally pulling a number through could cause side effects on downstream operations.

Reply:
This ID is set by us during creation of a TC, so it will be always a string, The type is provided by the echarts and so they have kept it as wide as possible...  I guess we could convert it instead of just casting... ```String(g.id)``` which may avoid run time error

Reply:
Got it, better to cast imo, doesn't have to be done in this PR.